src/test/regress/sql/multi_prepare_plsql.sql

```diff
@@ -555,6 +555,34 @@ $BODY$ LANGUAGE plpgsql;
 SELECT ddl_in_plpgsql();
 SELECT ddl_in_plpgsql();
 
+-- test prepared ddl with multi search path to make sure the schema name doesn't leak on
+-- to the cached statement
+CREATE OR REPLACE FUNCTION ddl_in_plpgsql2()
+RETURNS VOID  AS
+$BODY$
+BEGIN
+    CREATE INDEX prepared_index ON prepare_ddl(x);
+END;
+$BODY$ LANGUAGE plpgsql;
+
+CREATE SCHEMA otherschema;
+SET search_path TO otherschema, public;
+
+SELECT ddl_in_plpgsql2();
+DROP INDEX prepared_index;
+
+-- this creates the same table it 'otherschema'. If there is a leak the index will not be
+-- created on this table, but instead on the table in the public schema
+CREATE TABLE prepare_ddl (x int, y int);
+SELECT create_distributed_table('prepare_ddl', 'x');
+
+SELECT ddl_in_plpgsql2();
+-- list information for the index created to verify the schema of the table it was created
+-- for
+\d prepared_index
```

Comment:
There is `index_attrs` defined `src/test/regress/multi_test_helper.sql`. So, the following should also work
```SQL
SELECT nspname, relname, "Column", "Type" FROM index_attrs WHERE relname = 'prepare_ddl' ORDER BY 3,4;
```

Reply:
I used the query from marco, but select both the schemaname and the indexname. I didn't care about the columns the index was on, only on which schema it got created. The first commit shows it leaked onto the `public` schema, the fix made sure it was created on `otherschema`.

Also added some cleanup of the new objects created, and overwrote the function instead of the `...2` function name.